packages/liveblocks-client/src/types/index.ts

```diff
@@ -505,113 +519,41 @@ export type Room<
-      callback: (liveList: LiveList<TItem>) => void
-    ): () => void;
-
-    /**
-     * Subscribes to changes made on a {@link LiveMap} and all the nested data structures. Returns an unsubscribe function.
-     * In a future version, we will also expose what exactly changed in the {@link LiveMap}.
-     *
-     * @param callback the callback this called when the {@link LiveMap} changes.
-     *
-     * @returns Unsubscribe function.
-     *
-     * @example
-     * const liveMap = new LiveMap();
-     * const unsubscribe = room.subscribe(liveMap, (liveMap) => { }, { isDeep: true });
-     * unsubscribe();
-     */
-    <TKey extends string, TValue extends Lson>(
-      liveMap: LiveMap<TKey, TValue>,
-      callback: (updates: LiveMapUpdates<TKey, TValue>[]) => void,
-      options: { isDeep: true }
-    ): () => void;
-
-    /**
-     * Subscribes to changes made on a {@link LiveObject} and all the nested data structures. Returns an unsubscribe function.
-     * In a future version, we will also expose what exactly changed in the {@link LiveObject}.
-     *
-     * @param callback the callback this called when the {@link LiveObject} changes.
-     *
-     * @returns Unsubscribe function.
-     *
-     * @example
-     * const liveObject = new LiveObject();
-     * const unsubscribe = room.subscribe(liveObject, (liveObject) => { }, { isDeep: true });
-     * unsubscribe();
-     */
-    <TData extends LsonObject>(
-      liveObject: LiveObject<TData>,
-      callback: (updates: LiveObjectUpdates<TData>[]) => void,
-      options: { isDeep: true }
+    <L extends LiveStructure>(
+      liveStructure: L,
+      callback: (node: L) => void
     ): () => void;
 
     /**
-     * Subscribes to changes made on a {@link LiveList} and all the nested data structures. Returns an unsubscribe function.
-     * In a future version, we will also expose what exactly changed in the {@link LiveList}.
+     * Subscribes to changes made on a Live structure and all the nested data
+     * structures. Returns an unsubscribe function. In a future version, we
+     * will also expose what exactly changed in the Live structure.
      *
-     * @param callback the callback this called when the {@link LiveList} changes.
+     * @param callback The callback this called when the Live structure, or any
+     * of its nested values, changes.
      *
      * @returns Unsubscribe function.
      *
      * @example
-     * const liveList = new LiveList();
-     * const unsubscribe = room.subscribe(liveList, (liveList) => { }, { isDeep: true });
+     * const liveMap = new LiveMap();  // Could also be LiveList or LiveObject
+     * const unsubscribe = room.subscribe(liveMap, (updates) => { }, { isDeep: true });
      * unsubscribe();
      */
-    <TItem extends Lson>(
-      liveList: LiveList<TItem>,
-      callback: (updates: LiveListUpdates<TItem>[]) => void,
+    <L extends LiveStructure>(
+      liveStructure: L,
+      callback: StorageCallback,
```

Comment:
There's a fix here. When using `room.subscribe()` with `{ isDeep: true }` on, say, a LiveList, then `updates` will not be a `LiveListUpdate` array per se. It can also contain other updates, because `isDeep` will include all nested updates. This bug was pointed out by TypeScript correctly when I removed the `any`.

Reply:
And, for the record, the other bug that TypeScript pointed out here was fixed in https://github.com/liveblocks/liveblocks/pull/352/files#diff-c687f5217f04bdc741f7e57853e0c381ba3ffded12e1763d36dfb783b9ea642aR105 already.

Reply:
Can't believe that we missed those 2 typings issues...